Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java

```diff
@@ -0,0 +1,215 @@
+        }
+
+        String publisher = (String) payload.get("publisher");
+        if (publisher == null) {
+            throw new MissingAttributeException("Request body must contain 'publisher'");
+        }
+
+        String description = (String) payload.get("description");
+        if (description == null) {
+            throw new MissingAttributeException("Request body must contain 'description'");
+        }
+
+        Integer inventory = (Integer) payload.get("inventory");
+        if (inventory == null) {
+            throw new MissingAttributeException("Request body must contain 'inventory'");
+        }
+
+        String releaseDate = (String) payload.get("releaseDate");
+        if (releaseDate == null) {
+            throw new MissingAttributeException("Request body must contain 'releaseDate'");
+        }
+
+        String coverImage = (String) payload.get("coverImage");
+        if (coverImage == null) {
+            throw new MissingAttributeException("Request body must contain 'coverImage'");
+        }
+
+        Boolean isActive = true;
+
+        Listing listing = new Listing(
+                sellerUserId.longValue(),
+                isbn,
+                title,
+                floatPrice,
+                author,
+                publisher,
+                description,
+                inventory,
+                releaseDate,
+                isActive,
+                coverImage
+        );
+
+        repo.save(listing);
+
+        return listing;
+    }
+
+    @PutMapping("/listings")
+    public Map<String, Object> putListing(@RequestBody Map<String, Object> payload) {
+        Map<String, Object> map = new HashMap<>();
+
+        Integer listingId = (Integer) payload.get("listingId");
+        if (listingId == null) {
+            throw new MissingAttributeException("Request body must contain 'listingId'");
+        }
+
+        Listing listing = repo.findByListingId(listingId.longValue());
+
+        if (listing == null) {
+            throw new NotFoundException("Listing with id [" + listingId + "] cannot be found");
+        }
+
+        String isbn = (String) payload.get("isbn");
+        if (isbn != null) {
+            listing.setISBN(isbn);
+            map.put("isbn", isbn);
+        }
+        String title = (String) payload.get("title");
+        if (title != null) {
+            listing.setTitle(title);
+            map.put("title", title);
+        }
+
+        String price = (String) payload.get("price");
+        if (price != null) {
+            try {
+                Float floatPrice = Float.valueOf(price);
+                listing.setPrice(floatPrice);
+                map.put("price", price);
+            } catch (NumberFormatException e) {
+                throw new MissingAttributeException("Attribute 'price' must be formatted in decimal as a string (ex. '1.5')");
+            }
+            listing.setPrice(Float.valueOf(price));
+        }
```

Comment:
When I postman requested with 
{
...
"price":12.34
...
}
it throws a 500 error, saying Double can't be converted to string. With the way it is now, the float would have to be float2string'd before it gets sent to the API, and then gets string2float'd once received by the API. 

...or, use integer/long for price in cents....

Also add a check to make sure price isn't zero or negative

Reply:
They way I have it setup right now, price expects a string like "1.5". I improved the validation so now you should get better error messages instructing you how to use it correctly.

Reply:
So this is expecting a string of a float instead of a float directly? 

If we continue with this, make sure that the whoever ends up doing the create new listing functionality in the frontend is well aware of this fact